src/export/node_export_algorithm.rb

```diff
@@ -82,11 +87,98 @@ def run
+    static_groups_as_sets_of_nodes = static_groups.map do |triangles|
+      nodes = Set.new
+      triangles.each do |triangle|
+        nodes << triangle.first_node
+        nodes << triangle.second_node
+        nodes << triangle.third_node
+      end
+      nodes
+    end
+
+    rotary_hinges.each do |node_pair|
+      static_groups_pair = static_groups_as_sets_of_nodes.select { |node_set| node_set.include?(node_pair[0]) && node_set.include?(node_pair[1]) }
+
+      substructure_to_inset = choose_structure_to_inset static_groups_pair.to_a
+
+      node_pair.each do |node_to_inset|
+        other_node_to_inset = if node_pair[0] == node_to_inset
+                                node_pair[1]
+                              else
+                                node_pair[0]
+                              end
+
+        edges_from_hinge_into_inset = node_to_inset.incidents.select { |edge| substructure_to_inset.include? edge.other_node(node_to_inset) }.select { |edge| edge.opposite(node_to_inset) != other_node_to_inset }
+        nodes_to_reconnect_to = edges_from_hinge_into_inset.map { |edge| { :node =>  edge.opposite(node_to_inset), :type => edge.link_type} }
+
+        edges_from_hinge_into_inset.each(&:delete)
+
+        inset_vector = node_to_inset.position.vector_to(other_node_to_inset.position)
+        inset_vector.length = Configuration::DISTANCE_TO_INSET_ROTARY_HUBS
+        inset_position = node_to_inset.position + inset_vector
+
+        nodes_to_reconnect_to.each do |node_information|
+          Graph.instance.create_edge_from_points(inset_position, node_information[:node].position, link_type: node_information[:type])
+        end
+      end
+    end
+  end
+
+  def choose_structure_to_inset(static_groups_pair)
+    # 1. Criterion: Offset which has no pods
+    has_pods = static_groups_pair.map do |static_group|
+      static_group.to_a.any? { |node| node.pods.size > 0 }
+    end
+    return static_groups_pair[0] if !has_pods[0] && has_pods[1]
+    return static_groups_pair[1] if has_pods[0] && !has_pods[1]
+
+    # 2. Criterion: Number of nodes
+    if static_groups_pair[0].size > static_groups_pair[1].size
+      static_groups_pair[1]
+    else
+      static_groups_pair[0]
+    end
+  end
+
   def edge_angle(edge1, edge2)
     edge1.direction.normalize.dot(edge2.direction.normalize).abs
   end
 
-  # HACK: always choose the next edge, that has the minimum angle to the current
+  # Make sure that if static groups touch, they touch at exactly 2 points,
+  # otherwise, the structure will be bended, and not fabricateable with welding
+  # (statement yet to proof)
+  def check_for_only_simple_hinges(static_groups)
+    rotary_hinges = []
+    static_groups_as_sets_of_nodes = static_groups.map do |triangles|
+      nodes = Set.new
+      triangles.each do |triangle|
+        nodes << triangle.first_node
+        nodes << triangle.second_node
+        nodes << triangle.third_node
+      end
+      nodes
+    end
```

Comment:
Would be cool to factor out the calculation of `static_groups_as_sets_of_nodes` (~10 lines above) since it's used twice and would increase readability.